src/interactive/app/handlers.rs

```diff
@@ -247,33 +268,59 @@ impl AppState {
         if let Some(_entry) = traversal.tree.node_weight(index) {
             let path_to_delete = path_of(&traversal.tree, index);
             delete_directory_recursively(path_to_delete)?;
```

Comment:
I have a concern here. If `delete_directory_recursively` removes some nested entries but the fails, this line causes an early return. Then the subsequent line doing the cleanup of the traversal is not executed. Can this cause problems where the traversal does not match any more with the state of the filesystem?

Reply:
That's an interesting spot - I also took a closer look once more to understand what's going on there.

I agree that the view of the filesystem may not match anymore with the view we have of it in memory as we will keep the entry entirely even if only the last file fails to be deleted. This is done knowing that the user should be able to try again and that the entries pane will highlight non-existing files in red when browsing them.

> Can this cause problems where the traversal does not match any more with the state of the filesystem?

No, but the current implementation might be more surprising to some.

The _more_ proper way of doing this would certainly be to traverse the in-memory data structure and delete entries one by one after their corresponding file/directory on disk was deleted. Such functionality would have to deal with non-empty directories as somebody might have added files after due last ran, probably by deleting directories recursively even if our in-memory representation thinks it is empty already.

Maybe there is more that could be done here and I am absolutely open to your ideas or an attempt to improve the situation. Maybe that could be another PR though.